pkg/server/asset/cloud.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2015-present Oursky Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package asset
+
+import (
+	"errors"
+	"io"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/franela/goreq"
+)
+
+const (
+	cloudAssetSignerTokenRefreshInterval = 30 * time.Minute
+	cloudAssetSignerTokenExpiryInterval  = 2 * time.Hour
+)
+
+// CloudStore models the skygear cloud asset store
+type CloudStore struct {
+	appName   string
+	host      string
+	authToken string
+	urlPrefix string
+	public    bool
+
+	// signer related
+	signerToken              string
+	signerExtra              string
+	signerTokenExpiredAt     time.Time
+	signerTokenRefreshTicker *time.Ticker
+	signerTokenMutex         *sync.RWMutex
+}
+
+type refreshSignerTokenResponse struct {
+	Value     string    `json:"value"`
+	Extra     string    `json:"extra"`
+	ExpiredAt time.Time `json:"expired_at"`
+}
+
+// NewCloudStore creates a new cloud asset store
+func NewCloudStore(
+	appName string,
+	host string,
+	authToken string,
+	publicURLPrefix string,
+	privateURLPrefix string,
+	public bool,
+) (*CloudStore, error) {
+	if appName == "" {
```

Comment:
I think you could just panic... the reason being that passing wrong parameters are programming error and there is no way to handle such error.